### PR TITLE
Configurable CabSignal activation with cabin activation

### PR DESCRIPTION
**Pull request by Xoov**

### CODE REVIEW

File: McZapkie/Mover.cpp [hunk 1]
```diff
@@ -128,16 +128,22 @@ void TSecuritySystem::update(double dt, double vel, bool pwr) {
 		vigilance_timer = 0.0;
 		alert_timer = 0.0;
 		press_timer = 0.0;
+		if (!DebugModeFlag)
+			cabactive = 0;
 		return;
 	}
+	
+	bool is_powered = !power && pwr;
```
Comment (Milek7) [inline McZapkie/Mover.cpp:136]: confusing var name (`just_powered_on`?)

File: McZapkie/Mover.cpp [hunk 2]
```diff
@@ -128,16 +128,22 @@ void TSecuritySystem::update(double dt, double vel, bool pwr) {
...
+	bool is_cabactived = CabDependent && (cabactive != cab);
```
Comment (Milek7) [inline McZapkie/Mover.cpp:137]: confusing var name (`just_activated`?)

File: McZapkie/Mover.cpp [hunk 3]
```diff
@@ -128,16 +128,22 @@ void TSecuritySystem::update(double dt, double vel, bool pwr) {
 		vigilance_timer = 0.0;
 		alert_timer = 0.0;
 		press_timer = 0.0;
+		if (!DebugModeFlag)
```
Comment (Milek7) [inline McZapkie/Mover.cpp:131]: I think it's better to add condition to line 139 instead.

**Review by Milek7 — CHANGES_REQUESTED** [review]
Instead of this, couldn't we just cut pwr signal while cab is not active?

**Comment by Xoov** [comment]
> Instead of this, couldn't we just cut pwr signal while cab is not active?

Yes. However it will be changed when refactoring that code and adding new features (cutting off, true multi-traction support, emergency loop).

**Pull request merged**